hail/src/main/resources/include/hail/ApproximateQuantiles.h

```diff
@@ -0,0 +1,76 @@
+#include <algorithm>
+#include <array>
+#include <random>
+#include <vector>
+#include <iostream>
+
+template<int log2_buffer_size>
+class ApproximateQuantiles {
+  static constexpr int buffer_size = 1 << log2_buffer_size;
+  // std::random_device rd;
```

Comment:
was this used before `std::uniform_int_distribution`? Should delete I think

Reply:
patrick explained the seeding, carry on

Reply:
Agreed, I think we should delete.

Reply:
erm, I think we should fix it to actually use rd

Reply:
@patrick-schultz do you want this to be a parameter? so we can seed the randomness?

Reply:
@patrick-schultz aside, I think you should changes requested until you're happy with this to be merged

Reply:
We definitely want a way to seed the randomness. I think taking the `random_device` as a parameter is probably the right way to do that, but I'd have to dig into the standard libraries rng framework to be sure. Anyways, I wouldn't hold this PR up on details like this. They're easy to change later, and it's not like this is hooked up to anything yet.